arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1443,11 +1446,14 @@ Result RocksDBCollection::insertDocument(transaction::Methods* trx,
         }
       }
       auto& rIdx = basics::downCast<RocksDBIndex>(*it->get());
+
+      cct.setConcurrencyControl(rIdx.unique());
       // if we already performed the preflight checks,
       // there is no need to repeat the checks once again here
       res = rIdx.insert(*trx, mthds, documentId, doc, options,
                         /*performChecks*/ !performPreflightChecks);
       if (!res.ok()) {
+        cct.setConcurrencyControl(true);
```

Comment:
Don't you want to put `cct.setConcurrencyControl(rIdx.unique());` to the `reverseIdxOps` lambda?

Reply:
IMHO it is not important to optimize the reverse operation.
The reverse operation is only called in case something went wrong, and the insert is rolled back. This should be a rather exceptional situation for which we do not need to optimize performance.

Reply:
Agreed, but it doesn't make code more difficult I think, so why not?

Reply:
It is not difficult, but every further iteration for micro-optimizations cost time & money.

Reply:
Yeah copy and paste single line is cost a lot of money, especially compared to something like absolutely unnecessary renaming mthds to methods.

Ok, I don't think it's important, so I resolve it